dev.sh

```diff
@@ -0,0 +1,160 @@
+
+resync_hosted_skill_repo(){
+    git checkout dev
+    git pull --rebase
+    git merge master
+    # If everything is managed via the script, then we can just push (no conflicts)
+    git push --no-verify
+    git checkout master
+}
+
+# Check if the command is valid, commands can be:
+# -> new: Create a new blank target skill project as an Alexa hosted skill (requires configured ask-cli)
+# -> init <skill_id>: Initialize a target skill project with an existing skill id (Must be a custom alexa hosted skill)
+# -> list: List all the available target skill projects
+# -> update <skill_slug>: Update the skill hosted repo with the code from the local repo (deploy)
+# -> config <config_file_path>: Configure the target skill to be deployed using a copy of the provided config file
+# -> dialog <skill_slug> <locale>: Debug the dialog model for the target skill
+# -> debug <skill_slug>: Debug the code for the skill hosted repo
+
+COMMAND=$1
+
+case $COMMAND in
+  new)
+
+    mkdir -p build
+    mkdir -p build/hosted
+    cd build/hosted
+
+    ask new
+
+    echo "🔗 Finished. Current targets:"
+    ls
+    ;;
+
+  init)
+
+    if [ -z "$2" ]; then
+      echo "Usage: ./dev.sh init <skill-id>"
+      exit 1
+    fi
+    SKILL_ID=$2
+
+    mkdir -p build
+    mkdir -p build/hosted
+    cd build/hosted
+    pwd
+    ask init --hosted-skill-id $SKILL_ID
+
+    echo "🔗 Finished. Current targets:"
+    ls
+    ;;
+
+  update)
+    cd build/hosted
+    # Hosted build directory can be given as an argument, otherwise its $(ls -d */ | grep -v build | head -n 1)
+    HOSTED_BUILD_DIR=${2:-$(ls -d */ | grep -v build | head -n 1)}
+    cd $HOSTED_BUILD_DIR
+    echo "Updating hosted skill target repo: $HOSTED_BUILD_DIR"
+
+    update_hosted_skill_repo > /dev/null 2>&1
+
+    # Copy over the config file if it exists in ../<$HOSTED_BUILD_DIR>_config.json
+    if [ -f "../${HOSTED_BUILD_DIR}_config.json" ]; then
+      cp "../${HOSTED_BUILD_DIR}_config.json" "./lambda/config.json"
+    else
+      # Create one from the default config
+      cp "./lambda/config.json" "../${HOSTED_BUILD_DIR}_config.json"
+    fi
+
+    # Copy over the invocation name if it exists as "invocation_name" in ../<$HOSTED_BUILD_DIR>_config.json
+    INVOCATION_NAME=$(cat "../${HOSTED_BUILD_DIR}_config.json" | jq -r '.invocation_name')
+    if [ -z "$INVOCATION_NAME" ]; then
+      echo "Invocation name not found in the config file. Skipping invocation name update"
+    else
+      echo "Updating invocation name to $INVOCATION_NAME"
+      if [ "$(uname)" = "Darwin" ]; then
+        sed -i '' "s/\"invocationName\": \"[^\"]*\"/\"invocationName\": \"$INVOCATION_NAME\"/" skill-package/interactionModels/custom/en-US.json
+      else
+        sed -i "s/\"invocationName\": \"[^\"]*\"/\"invocationName\": \"$INVOCATION_NAME\"/" skill-package/interactionModels/custom/en-US.json
+      fi
+    fi
+
+    git add .
+    git commit -a -m "Trigger update from alexa-skill-llm-intent" --no-verify && git push
```

Comment:
Where are we going to be pushing too in this instance? wont this try and push to your branch if I run it?

Reply:
Yep. Inside build/hosted/ there's gonna be a folder per hosted skill. Because there are Alexa Hosted, they are versioned via git using Amazons own repos and branches scheme corresponding to their development / production environments and whatnot, with the full ask cli support.

So this is commiting in that versioned skill repo. We "init" it when we run `make new` or `make import id=<skill_id>`, that themselves run `ask cli` commands that take care of doing all the initialization.

Reply:
Just did the `make update` on a manually pushed one and saw  - thats quite neat!